File: romanToInteger.py
class Solution:
    def romanToInt(self, s: str) -> int:
        romanNum = {
            'I' : 1,
            'V' : 5,
            'X' : 10,
            'L' : 50,
            'C' : 100,
            'D' : 500,
            'M' : 1000
        }
        
        ans = 0

        for i in range(len(s)):
            if i < (len(s) - 1) and romanNum[s[i]] < romanNum[s[i + 1]]:
                ans -= romanNum[s[i]]
            else:
                ans += romanNum[s[i]]
        
        return ans

File: RomantoInteger.java
class Solution {
    public int romanToInt(String s) {
        Map<Character, Integer> num = new HashMap<Character, Integer>();
        num.put('I', 1);
        num.put('V', 5);
        num.put('X', 10);
        num.put('L', 50);
        num.put('C', 100);
        num.put('D', 500);
        num.put('M', 1000);

        int ans = 0;

        for (int i = 0; i < s.length(); i++){
            if (i < s.length() - 1 && num.get(s.charAt(i)) < num.get(s.charAt(i + 1))){
                ans = ans - num.get(s.charAt(i));
            }else{
                ans = ans + num.get(s.charAt(i));
            }
        }
        return ans;
    }
}
// Time: O(n)
// Space: O(1)
